pandas/formats/format.py

```diff
@@ -906,8 +927,59 @@ def get_col_type(dtype):
                          if x else '{}') for x in row]
             else:
                 crow = [x if x else '{}' for x in row]
+            if i < clevels and self.fmt.header and self.multicolumn:
+                # sum up columns to multicolumns
```

Comment:
Can you put the content contained in this if statement in a separate method on the class? Eg call it something like `_format_multicolumn(..)` ? Then you can give it a docstring with some explanation what exactly happens. And that will make the big for loop here a bit more comprehensible.